packages/cli/generators/relation/belongs-to-relation.generator.js

```diff
@@ -101,10 +111,32 @@ module.exports = class BelongsToRelationGenerator extends BaseRelationGenerator
     await sourceFile.save();
   }
 
-  getBelongsTo(className, relationName, fktype) {
+  getBelongsTo(
+    className,
+    relationName,
+    defaultRelationName,
+    sourceKeyName,
+    fktype,
+  ) {
+    // checks if relation name is customized
+    let relationDecorator = [
+      {
+        name: 'belongsTo',
+        arguments: [`() =>  ${className}`],
+      },
+    ];
+    // already checked if the relation name is the same as the source key before
+    if (defaultRelationName !== relationName) {
+      relationDecorator = [
+        {
+          name: 'belongsTo',
+          arguments: [`() =>  ${className}, {name: '${relationName}'}`],
```

Comment:
I am a bit confused about `defaultRelationName` and `sourceKeyName`, wouldn't they be the same? 
I find it hard to think of a user scenario that `defaultRelationName`, `sourceKeyName` and `relationName` are all different...

Reply:
Good point. I will list out the differences:

Before, the generator asked `foreign key`, `relationName` for **hasMany**.  
 and it only asked `relationName` for **belongTo** if `defaultRelationName` doesn't exist. It used `relationName` as the foreign key name (e.g `customerId`). Then it use the target model as the _real relation name_ (e.g `customer).

This is why how it dealt with **belongsTo** is problematic:
1. if the default foreign key already exists, it didn't prompt
2. if the default foreign key doesn't exists, it takes whatever it is, but the _real relation name_ is still the target model, and this is different from how the `belongsToAccessor` generates the default name. For example: if you type in `customer_id` as your foreign key, the generator would have `customer` as the default relation name. But for the `belongsToAccessor`, the default relation name is `customer_id` instead of `customer`. That's why users are getting `Cannot find type of undefined` errors. 

In order to allow custom fk and relation name for belongsTo, I added a new field `sourceKeyName` to represent the foreign key of belongsTo. And `relationName` to represent the _real relation name_. 

The line `defaultRelationName !== relationName` here is to check if the relationName is customized. If so, it should be specified in the `name` field, just like how we documented in our docs [Relation Metadata](https://loopback.io/doc/en/lb4/BelongsTo-relation.html#relation-metadata).

Therefore, it can have correct property:
```
  @belongsTo(() => Customer, {name: 'my_customer'}) // customized relation name
  customer_id: number; // customized fk
```